client/src/views/TeamSubmit.vue

```diff
@@ -0,0 +1,186 @@
+        >
+          Hier
+        </router-link>
+        findest Du die aktuellen Teams mit all ihren Mitgliedern.
+      </p>
+    </div>
+    <div
+      v-if="!loggedInUserIsAlreadyInATeam && !submitSuccessful"
+      data-cy="team-submit-panel"
+    >
+      <ul class="mb-3">
+        <li>Ein Team besteht immer aus 5 Mitgliedern</li>
+        <li>Der Teamname muss einmalig sein</li>
+        <li>Teammitglieder können immer nur einem Team angehören</li>
+      </ul>
+      <BaseInput
+        v-model="newTeamName"
+        label="Teamname"
+        :external-validation-result="nameIsAlreadyInUse"
+        data-cy="team-input-team"
+      />
+      <hr />
+      <BaseSelect
+        v-model="selectedUser"
+        label="Teammitglied auswählen"
+        :options="userNamesWithoutTeam"
+        data-cy="team-select-member"
+      />
+      <button
+        class="mb-3 btn btn-primary btn"
+        type="submit"
+        :disabled="!selectedUser || teamIsCompleted"
+        @click.prevent="onAddMember"
+      >
+        Mitglied hinzufügen
+      </button>
+      <ul class="list-group list-unstyled" data-cy="team-list-members">
+        <li>
+          <p>{{ submitterName }}</p>
+        </li>
+        <li v-for="member in newTeamMembers" :key="member">
+          <p>
+            <a href="#">
+              <i class="bi bi-trash" @click="onRemoveMember(member)"></i
+            ></a>
+
+            {{ member }}
+          </p>
+        </li>
+      </ul>
+      <button
+        class="mt-3 btn btn-primary btn"
+        type="submit"
+        :disabled="nameIsAlreadyInUse || !teamIsCompleted"
+        @click.prevent="onSubmit"
+      >
+        Teammeldung absenden <BaseSpinner v-if="showSpinner" />
+      </button>
+    </div>
+    <div
+      v-if="loggedInUserIsAlreadyInATeam && !submitSuccessful"
+      data-cy="team-submit-already-associated"
+    >
+      <p>
+        Du bist bereits in einem Team angemeldet. Deshalb kannst du kein neues
+        Team melden.
+        <router-link
+          :to="{
+            name: 'ResultsTeams',
+            params: { year: new Date().getFullYear() },
+          }"
+        >
+          Hier
+        </router-link>
+        findest Du die aktuellen Teams mit all ihren Mitgliedern.
+      </p>
+    </div>
+  </slot-dialog>
+</template>
+
+<script setup>
+import { computed, ref } from "vue";
+import useUser from "../composables/useUser";
+import ApiService from "../services/ApiService";
+
+const { authData, getUserId } = useUser();
+
+const submitterName = authData.value.firstName + " " + authData.value.lastName;
+const submitterId = getUserId.value;
+
+const newTeamName = ref("");
+const newTeamMembers = ref([]);
+const selectedUser = ref("");
+
+const teamNames = ref([]);
+const userNamesWithoutTeam = ref([]);
+const usersWithoutTeamData = ref([]);
+
+const showSpinner = ref(false);
+const submitSuccessful = ref(false);
+
+try {
+  const [teamNamesRes, usersWithoutTeamRes] = await Promise.all([
```

Comment:
Eventuell in der UI nach ergänzen, dass Piloten mit Team hier nicht angezeigt werden.

Reply:
done